R/main_processData.R

```diff
@@ -62,9 +89,32 @@ processData <- function(datasets, config){
   pooledStdDev <- calculatePoooledStdDev(calibratedDatasets)
   
   processedData <- accumulateMeasurementsForEachSample(calibratedDatasets, config)
-  
-  list(memoryCorrected = memoryCorrected,
-       calibrated = calibratedDatasets,
-       processed = processedData,
-       pooledStdDev = pooledStdDev)
-}
\ No newline at end of file
+
+  # fill output structure
+
+  namesOfDatasets <- names(datasets)
+
+  for (i in 1 : length(datasets)) {
+
+    output[[i]]$name <- namesOfDatasets[i]
+    
+    output[[i]]$raw <- datasets[[i]]
+    
+    output[[i]]$processed <- processedData[[i]]
+    
+  }
```

Comment:
This for loop could be replaced with the following code. If we do it this way we don't even need to initialize the output. (We could get rid of lines 41 to 65.)

```R
output <- map(names(datasets), function(name){
  list(
    name = name,
    raw = datasets[[name]],
    processed = processedData[[name]]
  )
})
```

Reply:
Thanks for the suggestion! If we do it this way, we need if-else constructions for the switchable processes, - such as memory correction: if (correction) fill with values else fill with NULL - to ensure the output structure to always be complete, right?

Reply:
That would be one good way to do it. Alternatively we can always set the default in the map and use an if construct to modify the output afterwards.